compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -2034,6 +2038,43 @@ public void exitStringTemplateLiteral(BallerinaParser.StringTemplateLiteralConte
         this.pkgBuilder.createStringTemplateLiteral(getCurrentPos(ctx), getWS(ctx), stringFragments, endingText);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void enterDocumentationAttachment(BallerinaParser.DocumentationAttachmentContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+        this.pkgBuilder.startDocumentationAttachment(getCurrentPos(ctx));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void exitDocumentationTemplateContent(BallerinaParser.DocumentationTemplateContentContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+        String contentText = ctx.docText() != null ? ctx.docText().getText() : null;
+        this.pkgBuilder.setDocumentationAttachmentContent(getCurrentPos(ctx), getWS(ctx), contentText);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void exitDocumentationTemplateAttributeDescription
+    (BallerinaParser.DocumentationTemplateAttributeDescriptionContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+        String attributeName = ctx.DocumentationTemplateAttributeEnd().getText();
+        String endText = ctx.docText() != null ? ctx.docText().getText().trim() : null;
```

Comment:
put empty string, if docText is null. Then you can avoid, null check in pkgBuilder level.

Reply:
+1